proxy/http/HttpSM.cc

```diff
@@ -2919,7 +2919,19 @@ HttpSM::tunnel_handler_server(int event, HttpTunnelProducer *p)
         tunnel.append_message_to_producer_buffer(p, reason, reason_len);
         }
       */
-      tunnel.local_finish_all(p);
+      // add fix code for caching incompleted chunked response case under some terrible network circumstances
+      // disable cache write for the malformed chunked response
+      if ((p->do_dechunking || p->do_chunked_passthru) && p->chunked_handler.truncation) {
+        DebugSM("http", "[%" PRId64 "] [HttpSM::tunnel_handler_server] disable cache write due to server truncation but still send "
+                        "partial response to client",
+                sm_id);
+        tunnel.abort_cache_write_finish_others(p);
```

Comment:
IMO, prefer `chain_abort_cache_write`  here and keep `local_finish_all` out of the if-statement.